### PR TITLE
Fix backup issue in network config modules and network action plugins common code refactor

### PR DESCRIPTION
**Pull request by ganeshrn**
##### SUMMARY

*  Fix `get_working_path` not found issue introduced due to
   backup-config code refactor (PR #50208)

*  Further refactor config related action plugins to minimize
   duplicate code

*  Remove unwanted config action plugins

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

module_utils/network/common/backup.py
plugins/action/aireos_config.py
plugins/action/aruba_config.py
plugins/action/asa_config.py
plugins/action/bigip_imish_config.py
plugins/action/ce_config.py
plugins/action/cnos_config.py
plugins/action/dellos10_config.py
plugins/action/dellos6_config.py
plugins/action/dellos9_config.py
plugins/action/edgeos_config.py
plugins/action/enos_config.py
plugins/action/eos_config.py
plugins/action/exos_config.py
plugins/action/fortios_config.py
plugins/action/ios_config.py
plugins/action/iosxr_config.py
plugins/action/ironware_config.py
plugins/action/junos_config.py
plugins/action/netconf_config.py
plugins/action/nos_config.py
plugins/action/nxos_config.py
plugins/action/onyx_config.py
plugins/action/slxos_config.py
plugins/action/sros_config.py
plugins/action/voss_config.py
plugins/action/vyos_config.py
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by IPvSean** [comment]
It works on Cisco IOS for me->

```
[student3@ansible ansible]$ ansible --version
ansible 2.8.0.dev0 (testing_PR50301 6e701e0499) last updated 2019/01/02 16:08:07 (GMT +000)
  config file = /home/student3/.ansible.cfg
  configured module search path = [u'/home/student3/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/student3/ansible/lib/ansible
  executable location = /home/student3/ansible/bin/ansible
  python version = 2.7.5 (default, Sep 12 2018, 05:31:16) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
[student3@ansible ansible]$
[student3@ansible ansible]$
[student3@ansible ansible]$
[student3@ansible ansible]$ cd ..
[student3@ansible ~]$ ls
ansible  demos  networking-workshop  tower_workshop
[student3@ansible ~]$ cd tower_workshop/
[student3@ansible tower_workshop]$ ls
group_vars  ios  junos  network_backup.yml  network_restore.yml  network_setup  network_time.yml  README.md
[student3@ansible tower_workshop]$ cd network_setup/
[student3@ansible network_setup]$ ls
clean_cli.yml  clean_rtr2.cfg  cli.yml  rtr2.cfg  setup.yml  templates  test.yml
[student3@ansible network_setup]$ ansible-playbook test.yml

PLAY [rtr2] ********************************************************************************************************************************************************

TASK [ios_config] **************************************************************************************************************************************************
ok: [rtr2]

PLAY RECAP *********************************************************************************************************************************************************
rtr2                       : ok=1    changed=0    unreachable=0    failed=0    skipped=0
```

the playbook is

```
[student3@ansible network_setup]$ cat test.yml
---
- hosts: rtr2
  gather_facts: false
  connection: network_cli
  tasks:
    - ios_config:
        src: rtr2.cfg
        #src: "/tmp/backup/2018-12-24@14:14:10/rtr2"
```

**Comment by IPvSean** [comment]
works on juniper as well!

```
[student3@ansible network_setup]$ ansible-playbook test_juniper.yml

PLAY [rtr3] ********************************************************************************************************************************************************

TASK [junos_config] ************************************************************************************************************************************************
ok: [rtr3]

PLAY RECAP *********************************************************************************************************************************************************
rtr3                       : ok=1    changed=0    unreachable=0    failed=0    skipped=0

[student3@ansible network_setup]$ sudo vi ~/.ansible.cfg
[student3@ansible network_setup]$ ls
backup  backup.yml  clean_cli.yml  clean_rtr2.cfg  cli.yml  rtr2.cfg  rtr3.cfg  setup.yml  templates  test_juniper.yml  test.yml
[student3@ansible network_setup]$ ansible-playbook test_juniper.yml -vvvv
ansible-playbook 2.8.0.dev0 (testing_PR50301 6e701e0499) last updated 2019/01/02 16:08:07 (GMT +000)
  config file = /home/student3/.ansible.cfg
  configured module search path = [u'/home/student3/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/student3/ansible/lib/ansible
  executable location = /home/student3/ansible/bin/ansible-playbook
  python version = 2.7.5 (default, Sep 12 2018, 05:31:16) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
Using /home/student3/.ansible.cfg as config file
setting up inventory plugins
host_list declined parsing /home/student3/networking-workshop/lab_inventory/hosts as it did not pass it's verify_file() method
script declined parsing /home/student3/networking-workshop/lab_inventory/hosts as it did not pass it's verify_file() method
auto declined parsing /home/student3/networking-workshop/lab_inventory/hosts as it did not pass it's verify_file() method
Parsed /home/student3/networking-workshop/lab_inventory/hosts inventory source with ini plugin
Loading callback plugin default of type stdout, v2.0 from /home/student3/ansible/lib/ansible/plugins/callback/default.pyc

PLAYBOOK: test_juniper.yml *****************************************************************************************************************************************
1 plays in test_juniper.yml

PLAY [rtr3] ********************************************************************************************************************************************************
META: ran handlers

TASK [junos_config] ************************************************************************************************************************************************
task path: /home/student3/tower_workshop/network_setup/test_juniper.yml:6
<35.183.110.79> attempting to start connection
<35.183.110.79> using connection plugin netconf
<35.183.110.79> local domain socket does not exist, starting it
<35.183.110.79> control socket path is /home/student3/.ansible/pc/15be63cb81
<35.183.110.79> connection to remote device started successfully
<35.183.110.79> local domain socket listeners started successfully
<35.183.110.79> ESTABLISH NETCONF SSH CONNECTION FOR USER: jnpr on PORT 22 TO 35.183.110.79
<35.183.110.79>
<35.183.110.79> local domain socket path is /home/student3/.ansible/pc/15be63cb81
<35.183.110.79> ESTABLISH LOCAL CONNECTION FOR USER: student3
<35.183.110.79> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/student3/.ansible/tmp/ansible-local-1722cdnj7K/ansible-tmp-1546446113.91-243008312592726 `" && echo ansible-tmp-1546446113.91-243008312592726="` echo /home/student3/.ansible/tmp/ansible-local-1722cdnj7K/ansible-tmp-1546446113.91-243008312592726 `" ) && sleep 0'
Using module file /home/student3/ansible/lib/ansible/modules/network/junos/junos_config.py
<35.183.110.79> PUT /home/student3/.ansible/tmp/ansible-local-1722cdnj7K/tmpvrsJQ7 TO /home/student3/.ansible/tmp/ansible-local-1722cdnj7K/ansible-tmp-1546446113.91-243008312592726/AnsiballZ_junos_config.py
<35.183.110.79> EXEC /bin/sh -c 'chmod u+x /home/student3/.ansible/tmp/ansible-local-1722cdnj7K/ansible-tmp-1546446113.91-243008312592726/ /home/student3/.ansible/tmp/ansible-local-1722cdnj7K/ansible-tmp-1546446113.91-243008312592726/AnsiballZ_junos_config.py && sleep 0'
<35.183.110.79> EXEC /bin/sh -c '/usr/bin/python /home/student3/.ansible/tmp/ansible-local-1722cdnj7K/ansible-tmp-1546446113.91-243008312592726/AnsiballZ_junos_config.py && sleep 0'
<35.183.110.79> EXEC /bin/sh -c 'rm -f -r /home/student3/.ansible/tmp/ansible-local-1722cdnj7K/ansible-tmp-1546446113.91-243008312592726/ > /dev/null 2>&1 && sleep 0'
ok: [rtr3] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "backup": false,
            "check_commit": false,
            "comment": "configured by junos_config",
            "confirm": 0,
            "confirm_commit": false,
            "host": null,
            "lines": null,
            "password": null,
            "port": null,
            "provider": null,
            "replace": null,
            "rollback": null,
            "src": "set version 17.4R1-S2.2\nset groups global system host-name rtr3\nset groups global system login user jnpr uid 2000\nset groups global system login user jnpr class super-user\nset groups global system login user jnpr authentication ssh-rsa \"ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDowy7eEePSBSf7VJ5HQScPKHpA1q+jQ6mWS1vvKZY7WZwS7YzO19C37shlUmCU/jnJ2aKBjeBmwXgLJGhpqQbMjH/tZ22Er4pKLQKK68Su7CvFk5yctPb3CENfz4ZL361a0TQ7J4GllaicpQb0kaPR/Kp3hZMSp/iMPb2QtXsruss+wljCfgVf6ZtYJQZPKYDJZABd8Z1O0bm09J14bPtTozmn/Ad9g+Rtj0hIDolfFWzqPgvwKR64VzgcVAvVRU+63xPfWT98vikWDU5zEyG++/iwbxPHt1X10BlzfCpP3ZqdDAKee2/RB1Q2gEZuKVKjhyrt6h/6OO2VM9HtXbu5 junostesting-key\"\nset groups global system services ssh root-login deny-password\nset groups global system syslog user * any emergency\nset groups global system syslog file messages any notice\nset groups global system syslog file messages authorization info\nset groups global system syslog file interactive-commands interactive-commands any\nset groups global interfaces fxp0 unit 0 family inet address 172.16.20.165/16\nset groups global routing-options static route 0.0.0.0/0 next-hop 172.16.0.1\nset groups global routing-options static route 0.0.0.0/0 retain\nset groups global routing-options static route 0.0.0.0/0 no-readvertise\nset apply-groups global\nset system root-authentication encrypted-password \"$6$Xow2/0WH$5ZoJDQ6rd46uC62tEPEMK13XXPTTlNCO0GRkLhVGg6ykOQSCAiHibENjuIlmkdLfek0ZOe904LPqAUtDFiFWP0\"\nset system services netconf ssh port 830\n",
            "src_format": null,
            "ssh_keyfile": null,
            "timeout": null,
            "transport": null,
            "update": "merge",
            "username": null,
            "zeroize": false
        }
    }
```

the playbook is simple
```
[student3@ansible network_setup]$ cat test_juniper.yml
---
- hosts: rtr3
  gather_facts: false
  connection: network_cli
  tasks:
    - junos_config:
        src: rtr3.cfg
        #src: "/tmp/backup/2018-12-24@14:14:10/rtr2"
```

**Review by NilashishC — APPROVED** [review]

**Comment by NilashishC** [comment]
LGTM @ganeshrn

**Review by trishnaguha — APPROVED** [review]

File: lib/ansible/plugins/action/network.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# (c) 2017, Red Hat, Inc.
+# (c) 2016 Red Hat Inc.
```
Comment (justjais) [inline lib/ansible/plugins/action/network.py]: should it not be `2018`.

Comment (ganeshrn) [reply lib/ansible/plugins/action/network.py]: Fixed

**Review by justjais — APPROVED** [review]

**Pull request merged**